### PR TITLE
Generalize search servlet

### PR DESCRIPTION
**Pull request by hajiler**
This is more so to see what I'm doing to talk about during pod meeting tomorrow. The biggest thing is displays a clickable class tree. It also extends the SearchServlet functionality to handle all orgs, keyword, as well as classes.

The problem to discuss during the pod meeting though is it relies on table with classifications as well as neighbors. So right now it hard codes neighbors for new CSV files, the idea is to add all new CSVs to same table with empty neighbors. Then later process knn nearest neighbors and update those columns.

### CODE REVIEW

File: website/src/main/webapp/results.html [hunk 1]
```diff
@@ -6,13 +6,16 @@
     <script src="prototype_script.js"></script>
   </head>
   
-  <body id="body" onload="addListener();">
+  <body id="body" onload="setUpPrototype();">
```
Comment (tonypan2000) [inline website/src/main/webapp/results.html:13]:
```suggestion
  <body id="body" onload="addListener(); setUpPrototype();">
```
addListener() adds listener for the enter keyup

File: website/src/main/java/com/google/step/data/CloudSQLManager.java [hunk 2]
```diff
@@ -109,4 +109,31 @@ public void createTable(String tableName, List<String> columns) throws SQLExcept
     }
   }
 
+  public ResultSet getOrgsWithNeighbors(String keyword, int offset) throws SQLException{ 
+    String similarTo = (!keyword.isEmpty()) ? 
+        "WHERE (name LIKE '%" + keyword + "%' OR about LIKE '% " + keyword + "%' OR class ='" + keyword + "')"
```
Comment (tonypan2000) [inline website/src/main/java/com/google/step/data/CloudSQLManager.java]: Optional: Should `class = keyword` or should `class LIKE '%keyword%'`?

File: website/src/main/java/com/google/step/data/CloudSQLManager.java [hunk 3]
```diff
@@ -109,4 +109,31 @@ public void createTable(String tableName, List<String> columns) throws SQLExcept
...
+    String query = String.join("\n","SELECT",
+            "preliminary.id, ",
+            "preliminary.link, ",
+            "preliminary.name, " ,
+            "preliminary.about, "  ,
+            "n1.name AS neighbor1, ",
+            "n2.name AS neighbor2, ",
+            "n3.name AS neighbor3, ",
+            "n4.name AS neighbor4",
+        String.format("FROM (%s) AS preliminary", preliminaryQuery),
```
Comment (tonypan2000) [inline website/src/main/java/com/google/step/data/CloudSQLManager.java:126]: sorry it's a bit difficult to understand, could you clarify what "preliminary" is?

File: website/src/main/java/com/google/step/data/OrganizationInfo.java [hunk 4]
```diff
@@ -68,12 +89,29 @@ public static OrganizationInfo getClassifiedOrgFrom(String[] record, int index)
...
+  }
+
   public void passInfoTo(PreparedStatement statement) throws SQLException {
     statement.setInt(1, this.id);
     statement.setString(2, this.name);
     statement.setString(3, this.link);
     statement.setString(4, this.about);
     String classPath = String.join("/", this.classification);
     statement.setString(5, classPath);
+    statement.setInt(6, 6);
```
Comment (tonypan2000) [inline website/src/main/java/com/google/step/data/OrganizationInfo.java]: Please fix hardcoded values before deploying

File: website/src/main/java/com/google/step/servlets/DataServlet.java [hunk 5]
```diff
@@ -64,7 +62,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         "name TEXT NOT NULL", 
         "link TEXT NOT NULL", 
         "about TEXT NOT NULL",
-        "class VARCHAR(255) NOT NULL");
+        "class VARCHAR(255) NOT NULL",
```
Comment (tonypan2000) [inline website/src/main/java/com/google/step/servlets/DataServlet.java]: Optional: Does VARCHAR(255) work for batches of 500 and more?

**Review by tonypan2000 — CHANGES_REQUESTED** [review]

File: website/src/test/java/com/google/step/DataServletTest.java [hunk 6]
```diff
@@ -19,101 +19,105 @@
...
-      Arrays.asList("Coding1","1Testing1","8Test2"),
-      Arrays.asList("Coding1","1Testing3","8Test3"),
-      Arrays.asList("Coding2","2Testing2","2Test"),
-      Arrays.asList("Coding2","2Testing2","2Test"),
-      Arrays.asList("Coding2","2Testing2","2Test"),
-      Arrays.asList("Coding3","3Testing3","3Test"),
-      Arrays.asList("Coding3","3Testing3","3Test"),
-      Arrays.asList("Coding3","3Testing4","3Test4"),
-      Arrays.asList("Coding3","3Testing4","3Test5"));
+  @Test
```
Comment (tonypan2000) [inline website/src/test/java/com/google/step/DataServletTest.java:22]: Please fix those tests after the demo tomorrow

**Review by tonypan2000 — APPROVED** [review]
LGTM for the demo

File: website/src/main/java/com/google/step/data/CloudSQLManager.java (hunk 2 above)
Comment (hajiler) [reply website/src/main/java/com/google/step/data/CloudSQLManager.java]: So I made a mistake when writing the classification code where it includes classifications with less accuracy. 
So something could be Health/Diseases/Cancer and People/Livilhood or something like that. I fixed it but until we redo the database using class = keyword won't yield anything

File: website/src/main/java/com/google/step/servlets/DataServlet.java (hunk 5 above)
Comment (hajiler) [reply website/src/main/java/com/google/step/servlets/DataServlet.java]: So far so good

**Pull request merged**